src/dns/mod.rs

```diff
@@ -0,0 +1,9 @@
+//! DNS resolution
+
+pub use resolve::{Addrs, Resolve, Resolving};
+pub(crate) use resolve::{DnsResolverWithOverrides, DynResolver};
+
+pub(crate) mod gai;
+pub(crate) mod resolve;
+#[cfg(feature = "trust-dns")]
+pub mod trust_dns;
```

Comment:
My reasoning was that it would be convenient to not have to re-implement the trust-dns resolver in client code. E.g., if users want to use the trust-dns resolver with custom configuration, they can import this module (this happens to be my specific use case). Though it wouldn't be a particularly big problem to re-implement the resolver. Thoughts?

Reply:
We've purposefully not exposed the specific `trust_dns` options and config types, since it then becomes a breaking change for us to try to upgrade internally. I've mentioned in other PRs that we _could_ expose generic DNS options that we translate for the trust-dns resolver. Otherwise, the best would be for someone to implement `Resolve` for their own newtype.

Reply:
Makes sense, I didn't think about that. Updated!